library-best-practices/JLBP-10.md

```diff
@@ -0,0 +1,33 @@
+[JLBP-10] Infrequent breaking changes
+-------------------------------------
+
+Every breaking change incurs a cost to the consumers. There is the immediate
```

Comment:
I'm not sure the cost is so high if older versions move forward in parallel branches. I think we need to rethink this JLBP while being explicit that there can be more than one branch.

Reply:
In that case, the cost is high for the maintainer, because now there are multiple branches (and releases) to maintain and keep track of. The cost is high for someone.

Reply:
Yes, the cost is higher for the maintainer; but is that a deal breaker? This documents best practices and I'm not surprised that the best practice here costs more. Quality and reliability aren't cheap. A maintainer may reasonably choose to prioritize cost for any given library, but I don't think we need to allow for that as a best practice.

Reply:
It's not merely cost; the cost is more complexity. I don't think we want to be in the business of pushing people to adopt more complex solutions if don't really, really have to.

Reply:
I think the benefits justify the cost, at least across the sum of consumers and suppliers. It is more complex than single repo, single branch, but I think it's a necessary one for libraries to not get locked into old APIs indefinitely.